lib/puppet/provider/image_streamer_artifact_bundle/image_streamer.rb

```diff
@@ -0,0 +1,105 @@
+################################################################################
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require_relative '../image_streamer_resource'
+
+Puppet::Type.type(:image_streamer_artifact_bundle).provide :image_streamer, parent: Puppet::ImageStreamerResource do
+  desc 'Provider for Image Streamer Artifact Bundles using the Image Streamer API'
+
+  mk_resource_methods
+
+  def exists?
+    artifacts_uri_parse('buildPlans')
+    artifacts_uri_parse('deploymentPlans')
+    artifacts_uri_parse('goldenImages')
+    artifacts_uri_parse('planScripts')
+    @deployment_group_class = OneviewSDK::ImageStreamer.resource_named('DeploymentGroup', @client.api_version)
+    super([nil, :found, :extract, :download])
+  end
+
+  def create
+    current_resource = @resourcetype.find_by(@client, unique_id).first
+    if current_resource
+      current_resource.update_name(@data['new_name']) if @data['new_name']
+    elsif @data['artifact_bundle_path']
+      @resourcetype.create_from_file(@client, @data['artifact_bundle_path'], @data['name'])
+    else
+      @resourcetype.new(@client, @data).create
+    end
+    true
+  end
+
+  def extract
+    artifact_bundle = get_single_resource_instance
+    artifact_bundle.extract
+    true
+  end
+
+  def download
+    path = @data.delete('artifact_bundle_download_path')
+    artifact_bundle = get_single_resource_instance
+    artifact_bundle.download(path)
```

Comment:
I think it makes sense in this case.
The previously content and the downloaded content can be compared only when the file is fully downloaded.